### PR TITLE
drivers: rcar_mmc: add unmap mmc memory and clk disable in case of er…

### PR DESCRIPTION
**Pull request by xakep-amatop**
…rors

Add unmap of SD/MMC device memory and SD/MMC clock stop in case of errors during initialization of the driver.

### CODE REVIEW

File: drivers/sdhc/rcar_mmc.c [hunk 1]
```diff
@@ -1707,12 +1707,13 @@ static int rcar_mmc_init(const struct device *dev)
...
 	if (ret < 0) {
 		LOG_ERR("%s: error can't apply pinctrl state", dev->name);
-		return ret;
+		goto _unmap;
 	}
 
 	if (!device_is_ready(cfg->cpg_dev)) {
 		LOG_ERR("%s: error cpg_dev isn't ready", dev->name);
-		return -ENODEV;
+		ret = -ENODEV;
```
Comment (LKomaryanskiy) [inline drivers/sdhc/rcar_mmc.c:1748]: (not your issue, minor): Looks like here we have to return -EBUSY instead of -ENODEV

Comment (xakep-amatop) [reply drivers/sdhc/rcar_mmc.c:1748]: unfortunately, in this file all issues mine :)
I'd just copy-pasted this check from some other driver long time ago, you may just search `if (!device_is_ready(` and look at how everyone handles this error

Comment (LKomaryanskiy) [reply drivers/sdhc/rcar_mmc.c:1748]: Yes, you are right, in most cases, in this case, functions return -ENODEV or less frequently  -EINVAL. So you can ignore my comment:) Thank you for your answer.

**Review by firscity — APPROVED** [review]
`Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>`

File: drivers/sdhc/rcar_mmc.c [hunk 2]
```diff
@@ -1762,6 +1763,15 @@ static int rcar_mmc_init(const struct device *dev)
 		dev->name, data->ver);
 
 	return 0;
+
+_disable_clk:
```
Comment (lorc) [inline drivers/sdhc/rcar_mmc.c]: This is not (probably) codified in the Zephyr coding style, but it is not customary to use underscores in the label names.  Your code will differ from the surrounding code. So I am strictly suggesting to rename your labels.

**Review by lorc — COMMENTED** [review]
With or without addressing my comment:

`Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>`

File: drivers/sdhc/rcar_mmc.c (hunk 2 above)
Comment (xakep-amatop) [reply drivers/sdhc/rcar_mmc.c]: changed names

**Comment by LKomaryanskiy** [comment]
`Acked-by: Leonid Komarianskyi <leonid_komarianskyi@epam.com>`

**Pull request merged**